development/build/index.js

```diff
@@ -164,89 +166,110 @@ function defineAndRunBuildTasks() {
-    throw new Error(`MetaMask build: Invalid build type: "${buildType}"`);
-  }
+test: Create an optimized build for running e2e tests.
 
-  const rawBuildVersion = argv[NamedArgs.BuildVersion];
-  const buildVersion = Number.parseInt(rawBuildVersion, 10);
-  if (rawBuildVersion.match(/^\d+$/u) === null || Number.isNaN(buildVersion)) {
-    throw new Error(
-      `MetaMask build: Invalid build version: "${rawBuildVersion}"`,
-    );
-  }
+testDev: Create an unoptimized, live-reloading build for debugging e2e tests.`,
+          type: 'string',
+        })
+        .option('apply-lavamoat', {
+          default: true,
+          description:
+            'Whether to use LavaMoat. Setting this to `false` can be useful during development if you want to handle LavaMoat errors later.',
+          type: 'boolean',
+        })
+        .option('build-type', {
+          default: BuildType.main,
+          description: 'The type of build to create.',
+          choices: Object.keys(BuildType),
+        })
+        .option('build-version', {
+          default: 0,
+          description:
+            'The build version. This is set only for non-main build types. The build version is used in the "prerelease" segment of the extension version, e.g. `[major].[minor].[patch]-[build-type].[build-version]``',
+          type: 'number',
+        })
+        .option('lint-fence-files', {
+          description:
+            'Whether files with code fences should be linted after fences have been removed. The build will fail if linting fails. This defaults to `false` if the entry task is `dev` or `testDev`. Otherwise this defaults to `true`.',
+          type: 'boolean',
+        })
+        .option('lockdown', {
+          default: true,
+          description:
+            'Whether to include SES lockdown files in the extension bundle. Setting this to `false` can be useful during development if you want to handle lockdown errors later.',
+          type: 'boolean',
+        })
+        .option('policy-only', {
+          default: false,
+          description:
+            'Stop the build after generating the LavaMoat policy, skipping any writes to disk other than the LavaMoat policy itself.',
+          type: 'boolean',
+        })
+        .option('skip-stats', {
+          default: false,
+          description:
+            'Whether to skip logging the time to completion for each task to the console. This is meant primarily for internal use, to prevent duplicate logging.',
+          hidden: true,
+          type: 'boolean',
+        })
+        .check((args) => {
+          if (!Number.isInteger(args.buildVersion)) {
+            throw new Error(
+              `Expected integer for 'build-version', got '${args.buildVersion}'`,
+            );
+          } else if (!Object.values(TASKS).includes(args.task)) {
+            throw new Error(`Invalid task: '${args.task}'`);
+          }
+          return true;
+        }),
+    )
```

Comment:
I wanted to use `.strict()` here, but it ended up breaking with LavaMoat.

Confusingly, the LavaMoat command uses the same process. There is no way to separate LavaMoat flags from flags being sent to the script being run by LavaMoat. Turning on strict mode would require adding all LavaMoat args here as parameters to ignore, or it'd require substantially changing how that LavaMoat binary works (e.g. add a way to... scrub LavaMoat options from `process.argv`, or a way to pass in new args for the script being run).

Reply:
Does `lavamoat` not support using `--` to distinguish LavaMoat options from the executable's options?

Reply:
Seemingly not. That would be a good solution to this. I have created an issue to track that idea: https://github.com/LavaMoat/LavaMoat/issues/344